File: gitdot-api/src/endpoint/review/list_reviews.rs
use serde::{Deserialize, Serialize};

use crate::{
    endpoint::Endpoint,
    resource::{common::Page, review::ReviewResource},
};

pub struct ListReviews;

impl Endpoint for ListReviews {
    const PATH: &'static str = "/repository/{owner}/{repo}/reviews";
    const METHOD: http::Method = http::Method::GET;

    type Request = ListReviewsRequest;
    type Response = ListReviewsResponse;
}

#[derive(ApiRequest, Debug, Default, Serialize, Deserialize)]
pub struct ListReviewsRequest {
    #[serde(default, skip_serializing_if = "Option::is_none")]
    pub cursor: Option<String>,
    #[serde(default, skip_serializing_if = "Option::is_none")]
    pub limit: Option<u32>,
}

pub type ListReviewsResponse = Page<ReviewResource>;

File: gitdot-web/packages/gitdot-api/src/endpoint/review/list-reviews.ts
import { z } from "zod";
import { page, ReviewResource } from "../../resource";
import type { Endpoint } from "../endpoint";

export const ListReviewsRequest = z.object({
  cursor: z.string().optional(),
  limit: z.number().int().positive().optional(),
});
export type ListReviewsRequest = z.infer<typeof ListReviewsRequest>;

export const ListReviewsResponse = page(ReviewResource);
export type ListReviewsResponse = z.infer<typeof ListReviewsResponse>;

export const ListReviews = {
  path: "/repository/{owner}/{repo}/reviews",
  method: "GET",
  request: ListReviewsRequest,
  response: ListReviewsResponse,
} as const satisfies Endpoint;
export type ListReviews = typeof ListReviews;
